src/stravalib/model.py

```diff
@@ -993,39 +991,44 @@ class ClubActivity(strava_model.ClubActivity):
     pass
 
 
-class DistributionBucket(TimedZoneRange):
+# Should be TimedZoneDistribution
+class TimedZoneDistribution(strava_model.TimedZoneRange):
     """
     A single distribution bucket object, used for activity zones.
 
     Notes
     -----
-    In this object, we override types for min/max values. The
-    strava API incorrectly types zones as being `int`. However it can
-    return `int` or `float`.
+    Min/max value types are overridden. The
+    Strava API incorrectly types zones as being `int`. However it can
+    return `int` or `float` (floats are returned for pace, ints for heartrate
+    and power).
     """
 
-    # Overrides due to a bug in the Strava API docs
-    # Because the created strava_model.py has this typed as int we will need
-    # to override these types
+    # Type overrides to support pace values returned as ints
     min: Optional[int | float] = None  # type: ignore
     max: Optional[int | float] = None  # type: ignore
 
 
-class BaseActivityZone(
-    ActivityZone,
+# TODO called ActivityZone in the API
+class ActivityZone(
+    strava_model.ActivityZone,
     BoundClientEntity,
 ):
     """
     Base class for activity zones.
 
-    A collection of :class:`stravalib.model.DistributionBucket` objects.
+    A collection of :class:`stravalib.strava_model.TimedZoneDistribution` objects.
     """
 
     # Field overrides from superclass for type extensions:
+    # TODO: what was the legacy behavior and is it ok to break it?
```

Comment:
@jsamoocha this is a sanity check. i will remove all comments once i'm clear this is correct. 

Is there legacy behavior that I need to consider here or can we just map to the API spec / returned values?

Reply:
We override the `min` and `max` attributes to also allow floats as their type. You already took care of that.